app/src/gui/fields/maps/MapFormField.tsx

```diff
@@ -48,11 +50,23 @@ export function MapFormField({
     initialFeatures = form.values[field.name];
   }
 
-  const [isAlert, setIsAlert] = useState(false);
+  // flag set if we find we don't have location permission
+  const [noPermission, setNoPermission] = useState(false);
 
   const [drawnFeatures, setDrawnFeatures] =
     useState<GeoJSONFeatureCollection>(initialFeatures);
 
+  useEffect(() => {
+    const perms = async () => {
+      const permission = await Geolocation.checkPermissions();
+      console.log('permission', permission);
+      if (permission.location === 'granted') {
+        await Geolocation.requestPermissions({permissions: ['location']});
+      }
```

Comment:
is this logic right? It's running request permissions if the location permission is granted?